persist/fs/files.go

```diff
@@ -232,6 +245,32 @@ func FilesetBefore(filePathPrefix string, namespace ts.ID, shard uint32, t time.
 	return filesBefore(matched, t)
 }
 
+// DeleteInactiveFilesets deletes any filesets that are not currently owned by the namespace
+func DeleteInactiveFilesets(filePathPrefix string, namespace ts.ID, activeShards []uint32) error {
+	var toDelete []string
+	dirs := make(map[string]bool)
+	activeShardDirs := activeShardDirs(filePathPrefix, namespace, activeShards)
+	namespaceDirPath := NamespaceDirPath(filePathPrefix, namespace)
+	allShardDirs, err := findDirectories(namespaceDirPath)
+	if err != nil {
+		return err
+	}
+
+	//can be extracted to a function perhaps?
+	for _, dir := range activeShardDirs {
+		dirs[dir] = true
+	}
+
+	//is there a general slice function that can handle this?
+	for _, dir := range allShardDirs {
+		ok := dirs[dir]
```

Comment:
To make this a little more robust perhaps we should compare the basename?  i.e. in case there's trailing slashes, or some symlink thing going on?
https://golang.org/pkg/path/#Base

Reply:
I use https://golang.org/pkg/os/#File.Readdirnames
which gives back the names of the dirs instead of the full path.

Reply:
I wanted to keep it simple by re-attaching the osPath to the dirnames, but I will remove the logic that handles that

Reply:
Actually, it might make sense to make a small struct of shard `uint32` and its associated directory